### PR TITLE
fix compiler warnings (except deprecation) and update gradle

### PR DESCRIPTION
**Pull request by smichel17**
TODO before merge:

- [ ] Check to make sure this doesn't break anything on dropbox version

### CODE REVIEW

**Comment by smichel17** [comment]
Of course you can decide to cherry-pick just the first commit if you want ;)

File: src/main/java/nl/mpcjanssen/simpletask/AddTask.kt [hunk 1]
```diff
@@ -192,9 +192,7 @@ class AddTask : ThemedActivity() {
         findViewById(R.id.btnDue)?.setOnClickListener { insertDate(DateType.DUE) }
         findViewById(R.id.btnThreshold)?.setOnClickListener { insertDate(DateType.THRESHOLD) }
 
-        if (selection != null && selection.size > 0) {
```
Comment (smichel17) [inline src/main/java/nl/mpcjanssen/simpletask/AddTask.kt]: Is always true.

**Comment by smichel17** [comment]
Android studio kept bugging me to update gradle, so I let it. Made no manual changes.

**Comment by mpcjanssen** [comment]
gradle update doesn't change anything for the app itself so no need to merge to master yet.

**Pull request merged**